lte/gateway/c/sctpd/util.cpp

```diff
@@ -102,8 +102,12 @@ int set_sctp_opts(
   event.sctp_association_event = on;
   event.sctp_shutdown_event    = on;
   event.sctp_data_io_event     = on;
+  int event_size               = sizeof(event);
+  if (event_size > 10) {
```

Comment:
This looks too hacky... Do we know why sizeof generates a wrong value?

Reply:
Agree that this is unexpected and a root cause would be good to know

Reply:
What kernel versions are running (`uname -a` if possible!) in each?

Reply:
I'm a bit confused actually - as linux kernel does include copies of `sctp_event_subscribe` - is that not the source of this struct? Meanwhile I unpacked `lksctp-tools-devel.x86_64` and found that `<netinet/sctp.h>` contains no struct definition for `sctp_event_subscribe`.

```
➜  lksctp-tools-devel-1.0.18-3.el8.x86_64 grep -R sctp_event_subscribe .
grep: ./usr/lib64/lksctp-tools/libwithsctp.so: No such file or directory
grep: ./usr/lib64/libsctp.so: No such file or directory
./usr/share/lksctp-tools/sctp_test.c:	struct sctp_event_subscribe subscribe;
./usr/share/lksctp-tools/sctp_darn.c:	struct sctp_event_subscribe subscribe;
```

Even though the package does supply netinet/sctp.h:

```
grep sctp_event_subscribe usr/include/netinet/sctp.h
```

Reply:
FWIW I dug into RHEL8 (kernel 4.18 and they claim upstream, so I assume [this is the defn of sctp_event_subscribe](https://sourcegraph.com/github.com/torvalds/linux@94710ca/-/blob/include/uapi/linux/sctp.h#L582:8), and also ubuntu 18.04 which appears to be kernel 5.3 though upstream less clear, so I looked at [this defn of sctp_event_subscribe](https://sourcegraph.com/github.com/torvalds/linux@4d856f7/-/blob/include/uapi/linux/sctp.h#L591:8)).  And in both cases, the struct has size 13 bytes...

Reply:
@rdefosse for visibility